addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -243,9 +243,11 @@ def partners(self, country=None, grade=None, page=0, **post):
             country_domain, ["id", "country_id"],
             groupby="country_id", orderby="country_id")
 
-        # Fallback: Show all partners when country has no associates.
+        # Fallback on all countries if no partners found for the country and
+        # there are matching partners for other countries.
         country_ids = [c['country_id'][0] for c in countries]
-        if country and country.id not in country_ids:
+        fallback_all_countries = country is not None and country.id not in country_ids
```

Comment:
Seems we don't have to modify the condition, do we ?
```suggestion
        fallback_all_countries = country and country.id not in country_ids
```